### ISSUE TITLE
Chapter 16: Modeling Sequential Data Using Recurrent

### ISSUE THREAD

**Issue by LasaniHussain** [rasbt/python-machine-learning-book-2nd-edition#109]
The RNN model gives a an accuracy of around 85% . Why is it so, even the statistical bag of words model gives accuracy around 85%. If it is so, then why are the advantages of RNN( learning sequential dependency) not reflected by an increase in accuracy.

**Comment by rasbt** [comment]
That's probably because it is a relatively small/simple dataset and RNN for illustration purposes. For the first edition, some readers were complaining that some code takes more than a few minutes to run, so we didn't want to add examples that run multiple hours (which could be the case with more complex datasets and RNNs). For deep learnign vs traditional machine learning, you typically have the following relationship:

<img width="645" alt="screen shot 2019-03-05 at 10 12 38 am" src="https://user-images.githubusercontent.com/5618407/53819292-48dd8280-3f2f-11e9-97c6-ef3f52e69f02.png">

Hope that helps :)

**Comment by LasaniHussain** [comment]
SIr,
I am using this IMDB dataset for my project work. So can you give
suggestion as how to improve the model for getting better accuracy with
larger/complex models on this same dataset since I need to illustrate to my
project guide that deep learning models at least give a better result than
statistical ones.
Warm regards,
Lasani.

**Comment by rasbt** [comment]
you can try to add an additional layer and/or increase the embedding and lstm sizes. However, it could potentially result in more overfitting since the dataset is relatively small.

**Comment by LasaniHussain** [comment]
SIr,
You have thoroughly explained the conversion to words to integer mapping
since integers are required to work with in RNN but it is not very clear
how each integer gets converted to embedding and what the embedding size
is. If each integer(word) gets converted to a fixed size embedding then how
are these embeddings fed into the stacked LSTM ?
I would be grateful if you clear this doubt of mine.

Warm regards,
Lasani.

**Comment by rasbt** [comment]
The embedding is basically a real vector (vector with real numbers) and you can determine the size. If you are familiar with the concept of hashing, this is basically a very similar approach. In other words, you take an integer representing your word and compute a vector with real numbers that represent the word. The network then uses this embedding vector for its further steps.

Maybe it helps to think of a multi-layer perceptron. the first layer is basically something like an embedding vector as it is computed form the input features

**Comment by LasaniHussain** [comment]
Sir,
I need to do cross validation while running training epochs in the code of
RNN for sentiment analysis. It can be easily done in keras but since your
code is in tensorflow, I don't know how to do this. Can you please suggest
something.
I would be really thankful if you could help me with this.

Warm regards,
Lasani.
